sklearn/linear_model/_passive_aggressive.py

```diff
@@ -445,6 +465,23 @@ class PassiveAggressiveRegressor(BaseSGDRegressor):
     [-0.02306214]
     """
 
+    _parameter_constraints = {
+        "loss": [StrOptions({"epsilon_insensitive", "squared_epsilon_insensitive"})],
+        "C": [Interval(Real, None, None, closed="neither")],
+        "fit_intercept": [bool],
+        "max_iter": [Interval(Integral, 1, None, closed="left")],
+        "tol": [Interval(Real, None, None, closed="neither"), None],
+        "shuffle": [bool],
+        "verbose": [Interval(Integral, 0, None, closed="left")],
+        "random_state": ["random_state"],
+        "early_stopping": [bool],
+        "validation_fraction": [Interval(Real, 0, 1, closed="neither")],
+        "n_iter_no_change": [Interval(Integral, 1, None, closed="left")],
+        "warm_start": [bool],
+        "average": [Interval(Integral, 0, None, closed="left"), bool],
+        "epsilon": [Interval(Real, 0, None, closed="left")],
+    }
```

Comment:
I think we should try to map out what parameters reside in what classes. A lot of this can be _greatly_ simplified via inheritance.  For example, `PassiveAggressiveRegressor` inherits from `BaseSGDRegressor`, but the parameter constraints here are separate. I think `BaseSGDRegressor` has the same constraints for all the parameters _except_ `[loss, C, epsilon]`.

This could then be simplified as 
```suggestion
    _parameter_constraints = {
        **BaseSGDRegressor._parameter_constraints,
        "loss": [StrOptions({"epsilon_insensitive", "squared_epsilon_insensitive"})],
        "C": [Interval(Real , 0, None, closed="neither")],
        "epsilon": [Interval(Real, 0, None, closed="left")],
    }
```